pkgs/by-name/gt/gtkclipblock/package.nix

```diff
@@ -0,0 +1,51 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, meson
+, cmake
+, ninja
+, pkg-config
+, gtk2
+, gtk3
+, gtk4
+}:
+
+let
+  distorm-src = fetchFromGitHub {
+    owner = "gdabah";
+    repo = "distorm";
+    rev = "3.5.2b";
+    hash = "sha256-2ftEV3TMS3HT7f96k+Pwt3Mm31fVEXcHpcbbz05jycU=";
+  };
+  source = fetchFromGitHub {
+    owner = "notpeelz";
+    repo = "gtkclipblock";
+    rev = "v0.2.2";
+    hash = "sha256-ok/D7M0KekN8zf8AzhcOLtedbYVRHHv3m9zEHsJfcPM=";
+    fetchSubmodules = true;
+  };
+in
+stdenv.mkDerivation {
+  pname = "gtkclipblock";
+  version = "0.2.2";
+
+  src = source;
+
+  nativeBuildInputs = [ meson cmake ninja pkg-config ];
+
+  buildInputs = [ gtk2 gtk3 gtk4 ];
+
+  postPatch = ''
+    substituteInPlace subprojects/funchook-helper/subprojects/funchook/CMakeLists.txt \
+      --replace "GIT_REPOSITORY https://github.com/gdabah/distorm.git" "SOURCE_DIR ${distorm-src}"
+  '';
+
+  dontUseCmakeConfigure = true;
+
+  meta = with lib; {
+    description = "A hack to prevent GTK programs from interacting with the primary clipboard";
```

Comment:
```suggestion
    description = "A hack to prevent GTK programs from interacting with the primary clipboard. To use it, prepend it to LD_PRELOAD";
```

Reply:
We don't - or shouldn't - do hints of usage in description.

Reply:
https://search.nixos.org/packages?query=LD_PRELOAD

Reply:
```suggestion
    description = "A LD_PRELOAD hack to prevent GTK programs from interacting with the primary clipboard";
```

Reply:
IMO one of the great strengths of nixpkgs is how much domain knowledge it contains and makes discoverable

Reply:
I'll let the maintainer, @UARTman, make the call :)

Reply:
Your second suggestion is way better.

We should not tell people how to use a software we package. 
Think on the absurdity of something like _"Emacs is a text editor. Use it to contribute to GitLab"_.